### PR TITLE
docs: change documentation on domain strategy

### PR DESCRIPTION
**Pull request by sousa-chinensis**
change full-config.md regarding https://github.com/p4gefau1t/trojan-go/pull/329

### CODE REVIEW

**Comment by sousa-chinensis** [comment]
Sorry that I have forgotten to modify the documentation,
so I tried my best to write something there in Chinese.

File: docs/content/basic/full-config.md [hunk 1]
```diff
@@ -228,11 +228,11 @@ weight: 30
 
 ```domain_strategy```域名解析策略，默认"as_is"。合法的值有：
 
-- "as_is"，只在域名列表中进行匹配。
+- "as_is"，只在各列表中的域名规则内进行匹配。
 
-- "ip_if_non_match"，在域名列表中进行匹配，如果不匹配，解析为IP后在IP列表中匹配。该策略可能导致DNS泄漏或遭到污染。
+- "ip_if_non_match"，先在各列表中的域名规则内进行匹配；如果不匹配则解析为IP后在各列表中的IP地址规则内进行匹配。该策略可能导致DNS泄漏或遭到污染。
```
Comment (Loyalsoldier) [inline docs/content/basic/full-config.md:233]:
```suggestion
- "ip_if_non_match"，先在各列表中的域名规则内进行匹配；如果不匹配，则解析为IP后，在各列表中的IP地址规则内进行匹配。该策略可能导致DNS泄漏或遭到污染。
```
```suggestion
- "ip_if_non_match"，先在各列表中的域名规则内进行匹配；如果不匹配则解析为IP后在各列表中的IP地址规则内进行匹配。该策略可能导致DNS泄漏或遭到污染。
```

File: docs/content/basic/full-config.md [hunk 2]
```diff
@@ -228,11 +228,11 @@ weight: 30
...
 
-- "ip_on_demand"，域名均解析为IP，在IP列表中匹配。该策略可能导致DNS泄漏或遭到污染。
+- "ip_on_demand"，先解析为IP后在各列表中的IP地址规则内进行匹配；如果不匹配则在各列表中的域名规则内进行匹配。该策略可能导致DNS泄漏或遭到污染。
```
Comment (Loyalsoldier) [inline docs/content/basic/full-config.md:235]:
```suggestion
- "ip_on_demand"，先解析为IP地址，在各列表中的IP地址规则内进行匹配；如果不匹配，则在各列表中的域名规则内进行匹配。该策略可能导致DNS泄漏或遭到污染。
```
```suggestion
- "ip_on_demand"，先解析为IP后在各列表中的IP地址规则内进行匹配；如果不匹配则在各列表中的域名规则内进行匹配。该策略可能导致DNS泄漏或遭到污染。
```

**Review by Loyalsoldier — CHANGES_REQUESTED** [review]

File: docs/content/basic/full-config.md [hunk 3]
```diff
@@ -228,11 +228,11 @@ weight: 30
...
 ```domain_strategy```域名解析策略，默认"as_is"。合法的值有：
 
-- "as_is"，只在域名列表中进行匹配。
+- "as_is"，只在各列表中的域名规则内进行匹配。
 
-- "ip_if_non_match"，在域名列表中进行匹配，如果不匹配，解析为IP后在IP列表中匹配。该策略可能导致DNS泄漏或遭到污染。
+- "ip_if_non_match"，先在各列表中的域名规则内进行匹配；如果不匹配，则解析为IP后，在各列表中的IP地址规则内进行匹配。该策略可能导致DNS泄漏或遭到污染。
 
-- "ip_on_demand"，域名均解析为IP，在IP列表中匹配。该策略可能导致DNS泄漏或遭到污染。
+- "ip_on_demand"，先解析为IP地址，在各列表中的IP地址规则内进行匹配；如果不匹配，则在各列表中的域名规则内进行匹配。该策略可能导致DNS泄漏或遭到污染。
```
Comment (Loyalsoldier) [inline docs/content/basic/full-config.md:235]:
```suggestion
- "ip_on_demand"，先解析为IP，在各列表中的IP地址规则内进行匹配；如果不匹配，则在各列表中的域名规则内进行匹配。该策略可能导致DNS泄漏或遭到污染。
```

**Review by Loyalsoldier — CHANGES_REQUESTED** [review]

**Pull request merged**